aeron-archive/src/main/java/io/aeron/archive/ArchiveConductor.java

```diff
@@ -2111,4 +2122,64 @@ private void deleteSegments(
             }
         }
     }
+
+    abstract static class Recorder extends SessionWorker<RecordingSession>
+    {
+        final MutableLong totalWriteBytes = new MutableLong();
+        final MutableLong totalWriteTimeNs = new MutableLong();
+        final MutableLong maxWriteTimeNs = new MutableLong();
+        private final Counter totalWriteBytesCounter;
+        private final Counter totalWriteTimeCounter;
+        private final Counter maxWriteTimeCounter;
+
+        Recorder(final CountedErrorHandler errorHandler, final Archive.Context context)
+        {
+            super("archive-recorder", errorHandler);
+            totalWriteBytesCounter = context.totalWriteBytesCounter();
+            totalWriteTimeCounter = context.totalWriteTimeCounter();
+            maxWriteTimeCounter = context.maxWriteTimeCounter();
+        }
+
+        public int doWork()
+        {
+            final int workCount = super.doWork();
+            if (workCount > 0)
+            {
+                totalWriteBytesCounter.set(totalWriteBytes.get());
```

Comment:
Should this be setOrdered as a release is only needed?

Reply:
Done.